packages/core/src/document/editor.tsx

```diff
@@ -29,81 +35,46 @@ export const DocumentEditor: React.FunctionComponent<
   const container = React.useRef<HTMLDivElement>(null)
 
-  const identifier = props.state
+  const identifier = props.state || createDocument()
   const { id } = identifier
 
   const store = React.useContext(EditorContext)
 
-  React.useEffect(() => {
-    if (!getDocument(store.state, id)) {
+  const deserializeHelpers: StoreDeserializeHelpers = {
+    createDocument(document: {
+      id: string
+      plugin?: string
+      state?: unknown
+    }): void {
       const plugin = getPlugin(
         store.state,
-        identifier.plugin || store.state.defaultPlugin
+        document.plugin || getDefaultPlugin(store.state)
       )
-
-      if (plugin && isStatefulPlugin(plugin)) {
-        const state = plugin.state(
-          deserializeDocument(identifier).state,
-          undefined,
-          () => {}
-        )
-
-        store.dispatch({
-          type: ActionType.Insert,
-          payload: {
-            ...identifier,
-            plugin: identifier.plugin || store.state.defaultPlugin,
-            state: state.$$value
-          }
-        })
-      } else {
-        store.dispatch({
-          type: ActionType.Insert,
-          payload: {
-            ...identifier,
-            plugin: identifier.plugin || store.state.defaultPlugin
-          }
-        })
-      }
       store.dispatch({
-        type: ActionType.ResetHistory
+        type: ActionType.Insert,
+        payload: {
+          id: document.id,
+          plugin: document.plugin || getDefaultPlugin(store.state),
+          ...(plugin && isStatefulPlugin<StateDescriptor<any>>(plugin)
+            ? {
+                state: document.state
+                  ? plugin.state.deserialize(document.state, deserializeHelpers)
+                  : plugin.state.createInitialState(deserializeHelpers)
+              }
+            : {})
+        }
       })
     }
-  }, [identifier])
-  const document = getDocument(store.state, id)
-  const [pluginState, setPluginState] = React.useState(null)
-
+  }
   React.useEffect(() => {
-    if (document) {
-      const plugin = getPlugin(store.state, document.plugin)
-
-      if (plugin && isStatefulPlugin(plugin)) {
-        const onChange = (param: unknown | ((value: unknown) => void)) => {
-          let stateHandler: (value: unknown) => unknown
-          if (typeof param === 'function') {
-            stateHandler = param as ((value: unknown) => unknown)
-          } else {
-            stateHandler = () => param
-          }
-
-          store.dispatch({
-            type: ActionType.Change,
-            payload: {
-              id,
-              state: stateHandler
-            }
-          })
-        }
-        const dispatch = (actions: Action[]) => {
-          R.forEach(store.dispatch, actions)
-        }
-        setPluginState(
-          plugin.state(identifier.state, document.state, onChange, dispatch)
-        )
-      }
+    if (!getDocument(store.state, id)) {
+      deserializeHelpers.createDocument(identifier)
```

Comment:
As we discussed: I'd move that up to `Editor` and fulfill the side effect in the store with an `Init` action.